### PR TITLE
CI job for running Python unit tests

### PR DESCRIPTION
**Pull request by cameel**
**NOTE: This PR is based on #10822. Please don't merge until that one is merged.**

This job runs the new unit tests introduced in #10675.

I'm running it also on Windows because I want to add some tests for #10676 that ensure that newlines are processed in the same way on CRLF platforms too.

EDIT: Turns out that the tests were not passing on Windows so I have rebased this PR on #10822 which fixes those issues.

### CODE REVIEW

**Comment by cameel** [comment]
Looks like bytecode comparison passed. I just pushed small tweaks now but they should not really change the result. This is now ready for review.

File: .circleci/config.yml
Comment (cameel) [inline .circleci/config.yml]: We could put this in `.gitattributes` but then it would also change for the build job and for contributors who work on Windows. I think that for the build job either is fine (with it enabled we're getting extra testing for CRLF versions) but not for users - we'd probably start getting CRLF endings in PRs. So I'm changing this only for specific jobs.

**Comment by cameel** [comment]
The command to run the Python tests locally on your machine (`PYTHONPATH="scripts/:$PYTHONPATH" python -m unittest discover --start-directory test/scripts/`) is pretty cumbersome so I added a wrapper script that does it. Now it can be executed from any directory and as a bonus the command is now identical on Windows (which was a bit tricky to get right).

I'm moving the PR back to draft for a moment because now that I pushed it, I need to resolve the CI failures.

**Review by hrkrshnn — DISMISSED** [review]
Looks good. Again one more review is nice before merging.

**Comment by cameel** [comment]
@hrkrshnn I just merged #10822 and looks like it dismissed your review :|

**Comment by cameel** [comment]
The change I just pushed is just a minor comment fix (`will executed` -> `will be executed`).

File: .circleci/config.yml
Comment (leonardoalt) [inline .circleci/config.yml:424]: Is `python3` for sure installed?

File: .circleci/config.yml
Comment (leonardoalt) [inline .circleci/config.yml:435]: Same here

File: test/pyscriptTests.py
Comment (leonardoalt) [inline test/pyscriptTests.py]: Should this make sure that python=python3?

File: .circleci/config.yml
Comment (hrkrshnn) [reply .circleci/config.yml:424]: It runs in the CI :) So, yes?

**Review by hrkrshnn — DISMISSED** [review]

File: .circleci/config.yml
Comment (leonardoalt) [reply .circleci/config.yml:424]: it's ubuntu so, you never know...

Comment (cameel) [reply .circleci/config.yml:424]: Yeah. It's actually the plain `python` executable that is not :) Initially I used `python` and I was getting errors saying that it does not exist.

On Ubuntu the `python` symlink is a part of some package other than `python` (it's in `python-pip` I think?) so if you just install Python 3, it's not there.

File: .circleci/config.yml
Comment (cameel) [reply .circleci/config.yml:435]: I haven't checked if `python3.exe` exists but `python.exe` does.

File: test/pyscriptTests.py
Comment (cameel) [reply test/pyscriptTests.py]: I don't think it's all that important (the script will most likely just fail on Python 2) but I can change it to use `python3` in the shebang.

Comment (cameel) [reply test/pyscriptTests.py]: Done.

File: .circleci/config.yml
Comment (cameel) [reply .circleci/config.yml:435]: Well, let's try and see. I changed that to `python3.exe` now.

**Review by leonardoalt — DISMISSED** [review]

**Comment by leonardoalt** [comment]
`python3.exe` fails right away though haha

**Comment by cameel** [comment]
Yeah :)
```
python3.exe : The term 'python3.exe' is not recognized as the name of a cmdlet, function, script file, or operable 
program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

Changed back.

**Review by leonardoalt — APPROVED** [review]

File: test/pyscriptTests.py
Comment (cameel) [reply test/pyscriptTests.py]: Nowadays they're the same on most systems I used plain `python` mostly because I was wary of a situation where `python` is available (and is Python 3) but `python3` is not. I haven't seen such a situation in practice though so, yeah, maybe let's use `python3` consistently.

File: .circleci/config.yml
Comment (cameel) [reply .circleci/config.yml:435]: No `python3.exe` :)

**Review by hrkrshnn — APPROVED** [review]

**Pull request merged**
